python/src/dataset.rs

```diff
@@ -564,7 +564,7 @@ impl Dataset {
 
     fn count_rows(&self) -> PyResult<usize> {
```

Comment:
Can probably also simplify the logic at https://github.com/lancedb/lance/blob/3c235deb41ad7113f0abf98249e0d4b96a7fe40b/python/python/lance/dataset.py#L614